larray/core.py

```diff
@@ -3784,6 +3789,143 @@ def _doc_agg_method(desc, by=False, action="perform",
+
+    def to_hdf(self, hdfstore, key):
+        if len(self):
+            hdfstore.get_storer(key).attrs.metadata = self
+
+    @classmethod
+    def from_csv(cls, filepath, sep=','):
+        def _convert_attribute(key, value):
+            # XXX : We could use the option 'parse_dates' of read_csv.
+            # Unfortunately, this option returns Series with only datetime or object dtype.
+            # We loose the automatic conversation of all other dtypes (bool, int, float).
+            # This is why we proceed in two steps bellow
+
+            # get Series with dtype bool, numeric (int, float) or object
+            ser = pd.read_csv(StringIO(key + '\n' + value), squeeze=True)
+            # convert object (string) to datetime when it's possible
+            if ser.dtypes == np.object_:
+                ser = pd.to_datetime(ser, errors='ignore', infer_datetime_format=True)
+            value = ser.values[0]
+            return key, value
+
+        attributes = _AttributeDict()
+        with open(filepath, 'r') as stream:
+            if stream.readline().strip() == '# Attributes':
+                for line in stream:
+                    if line.strip() == '# Data':
+                        break
+                    else:
+                        key, value = line.replace('#', '').strip().split(sep)
+                        key, value = _convert_attribute(key, value)
+                        attributes[key] = value
+        return attributes
+
+    # Use xlwings
+    @classmethod
+    def from_excel_xw(cls, sheet):
+        def _convert_value(value):
+            # Excel 'numbers' are always floats
+            if isinstance(value, float) and value == int(value):
+                value = int(value)
+            return value
+
+        attributes = _AttributeDict()
+        cell = sheet['A1']
+        if cell.value == 'Attributes':
+            while True:
+                cell = cell.offset(1, 0)
+                key, value = cell.value, cell.offset(0, 1).value
+                value = _convert_value(value)
+                if key == 'Data':
+                    break
+                attributes[key] = value
+        return attributes
+
+    # Use xlrd
+    @classmethod
+    def from_excel(cls, sheet):
+        def _convert_value(value, cell):
+            # ctype for cells (see doc of Cell class from xlrd):
+            # 2 --> XL_CELL_NUMBER, float
+            # 3 --> XL_CELL_DATE, float
+            # 4 --> XL_CELL_BOOLEAN, int (1 means TRUE, 0 means FALSE)
+            from xlrd import xldate_as_tuple
+            if cell.ctype == 2 and value == int(value):
+                value = int(value)
+            elif cell.ctype == 3:
+                value = datetime.datetime(*xldate_as_tuple(value, 0))
+            elif cell.ctype == 4:
+                value = bool(value)
+            return value
+
+        attributes = _AttributeDict()
+        rowx = 0
+        if sheet.cell(rowx=rowx, colx=0).value == 'Attributes':
```

Comment:
this is "dangerous". If, for any reason, you have an Attributes cell without a 'Data' cell, you could get an infinite loop (or a random Exception when trying to access an invalid cell). I usually try to avoid "while True:" unless the "breaking condition" is bullet-proof.